deepcell/training/model_trainer.py

```diff
@@ -0,0 +1,486 @@
+        self.model_path = model_path
+        self.model_version = model_version
+
+        # Add dataset information
+        self.X_train = X_train
+        self.y_train = y_train
+        self.X_test = X_test
+        self.y_test = y_test
+
+        # Add generator information
+        self.train_generator = train_generator
+        self.validation_generator = validation_generator
+        if isinstance(self.model.output_shape, list):
+            self.generator_skips = len(self.model.output_shape) - 1
+        else:
+            self.generator_skips = None
+        self.generator_seed = generator_seed
+        self.generator_batch_size = generator_batch_size
+        self.generator_transform = generator_transform
+        self.generator_transform_kwargs = generator_transform_kwargs
+        print(f'Transform kwargs: {self.generator_transform_kwargs}')
+        # Add miscellaneous information
+        self.dataset_metadata = dataset_metadata
+        self.postprocessing_fn = postprocessing_fn
+        print(self.postprocessing_fn)
+        self.postprocessing_kwargs = postprocessing_kwargs
+        self.predict_batch_size = predict_batch_size
+
+        # Add directories for logging and model export
+        if log_dir is None:
+            self.log_dir = os.path.join(model_path, 'logging')
+        else:
+            self.log_dir = log_dir
+
+        if tfserving_path is None:
+            self.tfserving_path = os.path.join(model_path, 'serving')
+        else:
+            self.tfserving_path = tfserving_path
+
+        # Add training kwargs
+        self.batch_size = training_kwargs.pop('batch_size', 1)
+        self.training_steps_per_epoch = training_kwargs.pop(
+            'training_steps_per_epoch',
+            None)
+        self.validation_steps_per_epoch = training_kwargs.pop(
+            'validation_steps_per_epoch',
+            None)
+        self.n_epochs = training_kwargs.pop('n_epochs', 8)
+        self.lr = training_kwargs.pop('lr', 1e-5)
+        self.lr_decay = training_kwargs.pop('lr_decay', 0.95)
+        self.lr_sched = training_kwargs.pop(
+            'lr_sched',
+            rate_scheduler(lr=self.lr, decay=self.lr_decay))
+        self.loss_function = training_kwargs.pop("loss_function",
+                "mean_squared_error")
+
+        # create optimizer
+        if "optimizer" in training_kwargs:
+            self.optimizer_passed = True
+            self.optimizer = training_kwargs.pop("optimizer")
+        else:
+            self.optimizer_passed = False
+            self.optimizer_learning_rate = 0.01
+            self.optimizer_decay = 1e-6
+            self.optimizer_momentum = 0.9
+            self.optimizer_nesterov = True
+            self.optimizer = SGD(
+                    lr= self.optimizer_learning_rate,
+                    decay= self.optimizer_decay,
+                    momentum= self.optimizer_momentum,
+                    nesterov= self.optimizer_nesterov)
```

Comment:
Maybe you can save a general `kwargs` for the optimizer, instead of specifically using a `_decay` and `_momentum` attribute.